modules/varnish/templates/default.vcl

```diff
@@ -197,6 +197,14 @@ sub mw_request {
 		if (req.method == "OPTIONS" && req.http.Origin) {
 			return (synth(200, "CORS Preflight"));
 		}
+		# From Wikimedia: https://gerrit.wikimedia.org/r/c/operations/puppet/+/120617/7/templates/varnish/upload-frontend.inc.vcl.erb
+		# required for Extension:MultiMediaViewer: T10285
+		if (req.url ~ "(?i)(\?|&)download(=|&|$)") {
+			/* Pretend that the parameter wasn't there for caching purposes */
+			set req.url = regsub(req.url, "(?i)(\?|&)download(=[^&]+)?$", "");
+			set req.url = regsub(req.url, "(?i)(\?|&)download(=[^&]+)?&", "\1");
+			set req.http.Content-Disposition = "attachment";
+		}

```

Comment:
Looks like you need to add:

```
if (req.http.X-Content-Disposition == "attachment") {
  set resp.http.Content-Disposition = "attachment";
}
```

somewhere from line 508?

Reply:
This is skipped in this version, while this is done in the Wikimedia version, this one just sets Content-Disposition directly. To be honest, I don't understand why this is done that way in Wikimedia, I think I'm just not understanding something about Varnish.

Reply:
Well this sends it to the backend the other one is responding to the client I think.

Reply:
> Well this sends it to the backend the other one is responding to the client I think.

Makes sense. I'll add the "if" to vcl_deliver.